src/oauth2-client/controller/collection.ts

```diff
@@ -73,6 +74,16 @@ class ClientCollectionController extends Controller {
       requirePkce: ctx.request.body.requirePkce ?? false,
     };
 
+    // Verify if client id already exists. Throw error if so.
+    try {
+      await findByClientId(clientId);
```

Comment:
The safer way to do this is to catch and check the error returned by the DB when doing the insert. In this scenario, two requests coming in at nearly the same time could both pass this check but one still fail the DB-level constraint.

Reply:
I agree, there could be other problems than 'Not Found' that shouldn't be translated into a `Conflict`.

Also it might be a bit better if this check is moved to inside the service.